src/main/java/user_interface/RecipeAppShell.java

```diff
@@ -1,4 +1,140 @@
+            System.out.println("Enter password: ");
+            String password = this.in.nextLine();
+            if (this.loginManager.login(username, password)) {
+                System.out.println("Login successful.");
+            } else {
+                System.out.println("Your username or password was incorrect.");
+            }
+        }
+    }
+
+    private void runLoggedInState() {
+        while(this.loginManager.getCurrUser() != null) {
+            System.out.println(this.getShellActions());
+            String action = this.in.nextLine();
+            this.runAction(this.getShellActionEnum(action));
+            System.out.println("You selected action " + action);
+        }
+    }
+
+    private ShellAction getShellActionEnum(String action) {
+        if (action.equals("0")) {
+            return ShellAction.BROWSEFEED;
+        } else if (action.equals("1")) {
+            return ShellAction.BROWSEPROFILE;
+        } else if (action.equals("2")) {
+            return ShellAction.POST;
+        } else if (action.equals("3")) {
+            return ShellAction.FOLLOW;
+        } else if (action.equals("4")) {
+            return ShellAction.CUSTOMIZEPROFILE;
+        } else if (action.equals("5")) {
+            return ShellAction.LOGOUT;
+        } else {
+            return ShellAction.INVALIDACTION;
+        }
+    }
+
+    private void runAction(ShellAction action) {
+        if (action == ShellAction.BROWSEFEED) {
+            // TODO: get Feed from FeedPresenter
+            System.out.println("This is your imaginary feed");
+        } else if (action == ShellAction.BROWSEPROFILE) {
+            // TODO: call UserProfilePresenter to display user profile info
+        } else if (action == ShellAction.POST) {
+            this.runPostCreation();
+        } else if (action == ShellAction.FOLLOW) {
+            this.runUserFollow();
+        } else if (action == ShellAction.CUSTOMIZEPROFILE) {
+            this.runProfileCustomize();
+        } else if (action == ShellAction.LOGOUT) {
+            this.runLogout();
+        } else {
+            System.out.println("That is not a valid action.");
+        }
+    }
+
+    private void runPostCreation() {
+        System.out.println("Enter your recipe title: ");
+        String title = this.in.nextLine();
+        while (true) {
+            System.out.println("Enter an ingredient name: ");
+            String ingredientName = this.in.nextLine();
+            System.out.println("Is your ingredient measurable? (Y/N)");
+            // TODO validate Y/N
+            String measureableAnswer = this.in.nextLine();
+            Boolean measurable = true;
+            System.out.println("Enter a quantity for your ingredient: ");
+            String quantity = this.in.nextLine();
+            Ingredient ingredient = new Ingredient();
+            System.out.println("Add another ingredient? (Y/N)");
+            String addIngredientAnswer = this.in.nextLine();
```

Comment:
Where can they enter the actual recipe?

Reply:
Sounds great, but how would the RecipeAppController be different from things like PostController and FeedController? Or can those classes split up tasks otherwise done by RecipeAppController?

Reply:
RecipeAppController would kind of be the "master" controller that always runs. If the user is logged out, it tells LoginPresenter to display the login page. If the user logs in through the inputs given to LoginPresenter, then LoginPresenter returns a boolean or something that indicates this to RecipeAppController. Then RecipeAppController would take some commands from the user input. For example, if the user told RecipeAppController "Post Recipe", then RecipeAppController would call PostPresenter.run(ShellAction.POST). Then PostPresenter would ask for inputs from the user by using the InOut interface.